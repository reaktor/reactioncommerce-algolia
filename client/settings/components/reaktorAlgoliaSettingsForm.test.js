import React from "react";
import renderer from 'react-test-renderer';
import ReaktorAlgoliaSettingsForm from "./reaktorAlgoliaSettingsForm";

const onChange = () => true;
const onSubmit = () => true;

test("ReaktorAlgoliaSettingsForm without settings", () => {
    const settings = {};
    const tree = renderer.create(
        <ReaktorAlgoliaSettingsForm settings={settings} onChange={onChange} onSubmit={onSubmit} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test("ReaktorAlgoliaSettingsForm with settings", () => {
    const settings = {
        "appId" : "appId",
        "adminApiKey" : "adminApiKey",
        "searchOnlyApiKey" : "searchOnlyApiKey",
        "productsIndex" : "productsIndex"
    };

    const tree = renderer.create(
        <ReaktorAlgoliaSettingsForm settings={settings} onChange={onChange} onSubmit={onSubmit} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
