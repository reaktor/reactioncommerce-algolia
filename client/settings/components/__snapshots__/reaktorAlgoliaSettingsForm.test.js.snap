// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReaktorAlgoliaSettingsForm with settings 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label>
      Algolia App Id:
      <input
        name="appId"
        onChange={[Function]}
        type="text"
        value="appId"
      />
    </label>
  </div>
  <div>
    <label>
      Algolia Admin API Key
      <input
        name="adminApiKey"
        onChange={[Function]}
        type="text"
        value="adminApiKey"
      />
    </label>
  </div>
  <div>
    <label>
      Algolia Search Only API Key
      <input
        name="searchOnlyApiKey"
        onChange={[Function]}
        type="text"
        value="searchOnlyApiKey"
      />
    </label>
  </div>
  <div>
    <label>
      Algolia Products Index Name
      <input
        label="Algolia Products Index Name"
        name="productsIndex"
        onChange={[Function]}
        type="text"
        value="productsIndex"
      />
    </label>
  </div>
  <button
    className="btn btn-primary pull-right"
    type="submit"
  >
    Save Changes
  </button>
</form>
`;

exports[`ReaktorAlgoliaSettingsForm without settings 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label>
      Algolia App Id:
      <input
        name="appId"
        onChange={[Function]}
        type="text"
      />
    </label>
  </div>
  <div>
    <label>
      Algolia Admin API Key
      <input
        name="adminApiKey"
        onChange={[Function]}
        type="text"
      />
    </label>
  </div>
  <div>
    <label>
      Algolia Search Only API Key
      <input
        name="searchOnlyApiKey"
        onChange={[Function]}
        type="text"
      />
    </label>
  </div>
  <div>
    <label>
      Algolia Products Index Name
      <input
        label="Algolia Products Index Name"
        name="productsIndex"
        onChange={[Function]}
        type="text"
      />
    </label>
  </div>
  <button
    className="btn btn-primary pull-right"
    type="submit"
  >
    Save Changes
  </button>
</form>
`;
